ServerLayer/Source/ServerLayer.h

```diff
@@ -1,287 +1,271 @@
+                                std::size_t, std::array<int, 3u>, std::array<int, 3u>, long long>
+        CONFIG;
+
+    /**
+     * \brief                        Constructor that initializes sockets, connects to server and
+     *                               creates socket to work with clients.
+     * \param[in] serverSendingPort  Server port to send.
+     * \param[in] serverRecivingPort Server port to recieve.
+     * \param[in] serverIP           Server IP address for connection.
+     * \param[in] layerPort          Additional port to communicate with clients.
+     * \param[in] backlog            The maximum length of the queue of pending connections.
+     * \param[in] workMode           Set work mode for layer to work with robot in safety
+     *                               or unsafety.
+     */
+    explicit ServerLayer(
+        const int serverSendingPort        = CONFIG.get<CAST(Param::DEFAULT_SENDING_PORT_TO_SERVER)>(),
+        const int serverRecivingPort       = CONFIG.get<CAST(Param::DEFAULT_RECEIVING_PORT_FROM_SERVER)>(),
+        const std::string_view serverIP    = CONFIG.get<CAST(Param::DEFAULT_SERVER_IP)>(),
+        const int layerPort                = CONFIG.get<CAST(Param::DEFAULT_CLIENT_PORT)>(),
+        const int backlog                  = 10,
+        const WorkMode workMode            = WorkMode::SAFE);
+
+    /**
+     * \brief			Deleted copy constructor.
+     * \param[in] other Other client object.
+     */
+                    ServerLayer(const ServerLayer& other)   = delete;
+
+    /**
+     * \brief			Deleted copy assignment operator.
+     * \param[in] other Other client object.
+     * \return			Returns nothing because it's deleted.
+     */
+    ServerLayer&    operator=(const ServerLayer& other)     = delete;
+
+    /**
+     * \brief			Deleted move constructor.
+     * \param[in] other Other client object.
+     */
+                    ServerLayer(ServerLayer&& other)        = delete;
+
+    /**
+     * \brief			Deleted move assignment operator.
+     * \param[in] other Other client object.
+     * \return			Returns nothing because it's deleted.
+     */
+    ServerLayer&    operator=(ServerLayer&& other)          = delete;
+
+    /**
+     * \brief Destructor that closes client socket.
+     */
+    virtual         ~ServerLayer();
+
+    /**
+     * \brief Main method which starts infinite working loop.
+     */
+    void            run() override;
+
+    /**
+     * \brief Process sockets.
+     */
+    void            launch() override;
+
+    /**
+     * \brief  Get server IP address.
+     * \return String which contains current server IP address.
+     */
+    std::string     getServerIP() const;
+
+    /**
+     * \brief                 Set server IP address.
+     * \param[in] newServerIP New server IP address as string.
+     */
+    void		    setServerIP(const std::string_view newServerIP);
+
+    /**
+     * \brief Additional fuction that receives data from server.
```

Comment:
**(must be changed, grammar nazi):** fu**n**ction

Reply:
Ok, I fixed it.